### ISSUE TITLE
Need a way to upgrade binary file formats

### ISSUE THREAD

**Issue by tjayrush** [Great-Hill-Corporation/quickBlocks#247]
I need to be able to upgrade binary file formats at will. It's built into the Serialize code already, but I don't know how to use it. I think an easy place to start this process is using the cacheMan code and the account caches.

Consolidate with #226

**Comment by tjayrush** [comment]
Existing binary files are in:
    other/acctIndex
    monitors
    ethslurp/tests.save/backup/slurps/
    .quickBlocks/slurps
    .quickBlocks/prices
    /Volumes/Samsung/scraper
         Indexes, miniBlocks, blocks, blooms

**Comment by tjayrush** [comment]
Referencing #226

**Comment by tjayrush** [comment]
SFArchive::m_archiveSchema should be renamed to m_writeSchema. This value should be used to assign to the class just before writing the data. If we're reading the data, update this value after reading. The only place the old value should ever exist is on the hard drive. Do not try to re-write the data unless you need to store modified data which for blocks will have to be retrieved from the node.

**Comment by tjayrush** [comment]
Closing because referenced by #226, but still active.

**Comment by tjayrush** [comment]
This was resolved in the ./other/blockCheck code where we successfully upgraded blocks, transactions, and receipts. The code in newblock.cpp, newtransaction.cpp, and newreceipt.cpp can be used as examples. Also, when we actually go to upgrade, we will create a frozen CBlock_513 (or whatever version we are upgrading from) to handle the reading from binary because, if we don't, and one of the new class's data types have changed, that would be a problem.
